Updating quantization documentation

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68907

Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
<img width="1197" alt="Screen Shot 2021-12-06 at 10 42 47 PM" src="https://user-images.githubusercontent.com/39544797/144962190-a835349a-9455-40eb-afed-38773a149382.png">
(full mathematical description of quantization process)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

HDCharles committed:
Updating quantization documentation

Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540

Test Plan:

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540 also made it so that doc uses torch.ao.quantization

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
![Screen Shot 2021-12-03 at 7 29 05 AM](https://user-images.githubusercontent.com/39544797/144602656-1d8d582a-5b84-42b7-8c20-e4ab329e9822.png)
(link works)

![Screen Shot 2021-12-03 at 7 29 41 AM](https://user-images.githubusercontent.com/39544797/144602727-559b6069-11e2-48ba-9872-bb3b1dc093b7.png)
(links changed to torch.ao.quantization)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

vkuzo commented:
thoughts about reverting the `ao` change for now?  The refactor to `torch.ao` is not user facing yet as we haven't finished the migration and havent published the OSS summary of what's happening.

vkuzo commented on docs/source/quantization-support.rst:
```diff
@@ -440,7 +440,10 @@ Note that operator implementations currently only
 operators. Furthermore the minimum and the maximum of the input data is
 mapped linearly to the minimum and the maximum of the quantized data
-type such that zero is represented with no quantization error.
+type such that zero is represented with no quantization error. This is slightly altered
+for symmetric qschemes, which instead use the negative and positive of the maximum absolute
+value of the input data in place of the normal minimum and maximum for the input data. For more info see how
+scale and zero point are calculated by :class:`~torch.ao.quantization.observer.MinMaxObserver`.
```
this makes sense.  Thoughts about just writing out the formulas in addition to the text, to remove even more ambiguity?

HDCharles replied:
ya, good idea

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540 also made it so that doc uses torch.ao.quantization

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
![Screen Shot 2021-12-03 at 7 29 05 AM](https://user-images.githubusercontent.com/39544797/144602656-1d8d582a-5b84-42b7-8c20-e4ab329e9822.png)
(link works)

![Screen Shot 2021-12-03 at 7 29 41 AM](https://user-images.githubusercontent.com/39544797/144602727-559b6069-11e2-48ba-9872-bb3b1dc093b7.png)
(links changed to torch.ao.quantization)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540 also made it so that doc uses torch.ao.quantization

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

vkuzo commented on docs/source/quantization-support.rst:
```diff
@@ -1,7 +1,7 @@
 -------------------------------
 
-torch.quantization
+torch.ao.quantization
```
should we remove the `ao` for now, since the migration is not user facing yet?

HDCharles replied:
done

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540 also made it so that doc uses torch.ao.quantization

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
<img width="1197" alt="Screen Shot 2021-12-06 at 10 42 47 PM" src="https://user-images.githubusercontent.com/39544797/144962190-a835349a-9455-40eb-afed-38773a149382.png">
(full mathematical description of quantization process)

<img width="1184" alt="Screen Shot 2021-12-06 at 10 42 20 PM" src="https://user-images.githubusercontent.com/39544797/144962151-b66294b9-04d7-4564-8eee-f39f99d7f499.png">
(note: torch.quantization->torch.ao.quantization)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540 also made it so that doc uses torch.ao.quantization

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
<img width="1197" alt="Screen Shot 2021-12-06 at 10 42 47 PM" src="https://user-images.githubusercontent.com/39544797/144962190-a835349a-9455-40eb-afed-38773a149382.png">
(full mathematical description of quantization process)

<img width="1184" alt="Screen Shot 2021-12-06 at 10 42 20 PM" src="https://user-images.githubusercontent.com/39544797/144962151-b66294b9-04d7-4564-8eee-f39f99d7f499.png">
(note: torch.quantization->torch.ao.quantization)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
<img width="1197" alt="Screen Shot 2021-12-06 at 10 42 47 PM" src="https://user-images.githubusercontent.com/39544797/144962190-a835349a-9455-40eb-afed-38773a149382.png">
(full mathematical description of quantization process)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

HDCharles committed:
Update on "Updating quantization documentation"


Summary: Added information about symmetric
qschemes and corrected an error in reference to https://github.com/pytorch/pytorch/issues/68540

Test Plan: cd docs
make html
python -m http.server
// open webpage, inspect it, looks good
<img width="1197" alt="Screen Shot 2021-12-06 at 10 42 47 PM" src="https://user-images.githubusercontent.com/39544797/144962190-a835349a-9455-40eb-afed-38773a149382.png">
(full mathematical description of quantization process)

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D32662033](https://our.internmc.facebook.com/intern/diff/D32662033)

[ghstack-poisoned]

HDCharles commented:
@HDCharles has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32662033).

vkuzo reviewed (approved):
Thanks for doing this!
